### PR TITLE
Add rcmwidget ads extension

### PR DESCRIPTION
**Pull request by yulya9060**
Add rcmwidget ads extension

Closes https://github.com/ampproject/amphtml/issues/33976

### CODE REVIEW

File: ads/vendors/rcmwidget.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
```
Comment (powerivq) [inline ads/vendors/rcmwidget.js:2]: 2021

File: ads/vendors/rcmwidget.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<!---
+Copyright 2017 The AMP HTML Authors. All Rights Reserved.
```
Comment (powerivq) [inline ads/vendors/rcmwidget.md:2]: 2021

**Review by powerivq — CHANGES_REQUESTED** [review]
LGTM except these comments

File: ads/vendors/rcmwidget.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import {loadScript, validateData} from '../../3p/3p';
+
+const WIDGET_DEFAULT_NODE_ID = 'rcm-widget';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function rcmwidget(global, data) {
+  validateData(data, [
```
Comment (powerivq) [inline ads/vendors/rcmwidget.js:26]: Nit: https://app.circleci.com/pipelines/github/ampproject/amphtml/7472/workflows/5d1a639a-0245-4fc2-bd30-b909bd55be58/jobs/109883

**Review by powerivq — CHANGES_REQUESTED** [review]

File: 3p/integration-lib.js [hunk 4]
```diff
@@ -96,6 +97,7 @@ const defaultAllowedTypesInCustomFrame = [
   'facebook',
   'twitter',
   'doubleclick',
+  'rcmwidget',
```
[the remark this answers is not in the document]
Comment (yulya9060) [reply 3p/integration-lib.js]: remove

File: ads/vendors/rcmwidget.js (hunk 1 above)
Comment (yulya9060) [reply ads/vendors/rcmwidget.js]: fix

File: ads/vendors/rcmwidget.md (hunk 2 above)
Comment (yulya9060) [reply ads/vendors/rcmwidget.md]: fix

File: ads/vendors/rcmwidget.js (hunk 3 above)
Comment (yulya9060) [reply ads/vendors/rcmwidget.js]: fix again)

**Review by powerivq — APPROVED** [review]

**Comment by powerivq** [comment]
@yulya9060 Could you rebase your branch against the latest main branch? It looks like something missing is preventing your PR from being tested.

**Comment by yulya9060** [comment]
> @yulya9060 Could you rebase your branch against the latest main branch? It looks like something missing is preventing your PR from being tested.

done

**Pull request merged**